contracts/commands/BasicSellCommand.sol

```diff
@@ -0,0 +1,133 @@
+    }
+
+    function isTriggerDataValid(uint256 _cdpId, bytes memory triggerData)
+        external
+        pure
+        returns (bool)
+    {
+        BasicSellTriggerData memory decodedTrigger = decode(triggerData);
+
+        (uint256 lowerTarget, ) = decodedTrigger.targetCollRatio.bounds(decodedTrigger.deviation);
+        return (_cdpId == decodedTrigger.cdpId &&
+            decodedTrigger.triggerType == 4 &&
+            decodedTrigger.execCollRatio < lowerTarget);
+    }
+
+    function isExecutionLegal(uint256 cdpId, bytes memory triggerData)
+        external
+        view
+        returns (bool)
+    {
+        BasicSellTriggerData memory decodedTriggerData = decode(triggerData);
+
+        (, uint256 nextCollRatio, uint256 nextPrice, ) = getVaultAndMarketInfo(cdpId);
+
+        return (decodedTriggerData.execCollRatio.wad() > nextCollRatio &&
+            decodedTriggerData.minSellPrice < nextPrice);
+    }
+
+    function execute(
+        bytes calldata executionData,
+        uint256 cdpId,
+        bytes memory triggerData
+    ) external {
+        BasicSellTriggerData memory decodedTriggerData = decode(triggerData);
+
+        validateTriggerType(decodedTriggerData.triggerType, 4);
+        validateSelector(MPALike.decreaseMultiple.selector, executionData);
+
+        executeMPAMethod(executionData);
+
+        if (decodedTriggerData.continuous) {
+            recreateTrigger(cdpId, decodedTriggerData.triggerType, triggerData);
+        }
+    }
+
+    function getDustLimit(bytes32 ilk) internal view returns (uint256 dustLimit) {
+        VatLike vat = VatLike(serviceRegistry.getRegisteredService(MCD_VAT_KEY));
+        (, , , , uint256 radDust) = vat.ilks(ilk);
+        uint256 wadDust = (radDust * WAD) / RAD;
+        return wadDust;
+    }
+
+    function getVaultInfo(uint256 cdpId) internal view returns (uint256 dustLimit) {
+        McdView mcdView = McdView(serviceRegistry.getRegisteredService(MCD_VIEW_KEY));
+        (, uint256 debt) = mcdView.getVaultInfo(cdpId);
+        return debt;
+    }
+
+    function isExecutionCorrect(uint256 cdpId, bytes memory triggerData)
+        external
+        view
+        returns (bool)
+    {
+        BasicSellTriggerData memory decodedTriggerData = decode(triggerData);
+        (, uint256 nextCollRatio, uint256 nextPrice, bytes32 ilk) = getVaultAndMarketInfo(cdpId);
+
+        uint256 dust = getDustLimit(ilk);
+
+        (, uint256 maxUnconditionallyAcceptedDebt) = dust.bounds(decodedTriggerData.deviation);
+
+        uint256 vaultDebtAfter = getVaultInfo(cdpId);
+
+        (uint256 lowerTarget, uint256 upperTarget) = decodedTriggerData.targetCollRatio.bounds(
+            decodedTriggerData.deviation
+        );
+
+        return
+            ((nextCollRatio > lowerTarget.wad() && nextCollRatio < upperTarget.wad()) ||
+                (vaultDebtAfter <= maxUnconditionallyAcceptedDebt)) &&
+            (nextPrice > decodedTriggerData.minSellPrice);
```

Comment:
this condition is wrong. please, make sure the parentheses are correctly put

Reply:
what You see incorrect here?

`(nextCollRatio > lowerTarget.wad() && nextCollRatio < upperTarget.wad())` = 'final collRation in right bounds'

|| = OR

`(vaultDebtAfter <= maxUnconditionallyAcceptedDebt)`  = we are near dustLimit 


ADDITIONALLY   = external brackets then && 

require that next price is legal

Reply:
I agree that `(nextPrice > decodedTriggerData.minSellPrice)` is redundant, I still need to think about the other part

Reply:
Other part is IMHO not needed and even incorrect. 

First BasicSell triggers only if nextPrice < currentPrice so same coll ratio 

second, let's even assume for a second that execution lagged for an hour or went to RPC and was executed later when oracle price changed 


If boundries fail and we are checking dust condition as acceptable excuse why we should reject it if nextPrice is > than currentPrice dust limit is computed based on currentPrice and user got his vault deleveraged and in nextPrice his collRatio will bounce even higher, what is wrong with that ?

Reply:
having `maxUnconditionallyAcceptedDebt` by itself is dangerous, because technically we could just always sell to the dust limit. we need to come up with additional condition that will sanity check this one. i agree that coll ratio comparison might not be a good fit here